ksa_compliance/utils/xml_parser.py

```
@@ -0,0 +1,272 @@
+import xml.etree.ElementTree as ET
+
+
+def parse_xml_to_dict(xml_string):
+    NAMESPACES = {
+        "": "urn:oasis:names:specification:ubl:schema:xsd:Invoice-2",
+        "cac": "urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2",
+        "cbc": "urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2",
+        "ext": "urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2",
+    }
+
+    root = ET.fromstring(xml_string)
```

Ruff (0.15.10)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)